src/extension.ts

```diff
@@ -19,6 +19,18 @@ export function activate(context: vscode.ExtensionContext) {
 	fs.mkdirSync(mapPath, { recursive: true } );
 	config.update('workspace.library', [ apiPath, mapPath ], false );
 
+	// Set the default diagnostic setting for the base Lua extension.
+	// Otherwise we get tons of "Undefined global" warnings for actors defined in map.yaml files.
+	// Only set it if no value is set to it at any level, and set only for the current workspace.
+	var baseLuaExtensionConfiguration = vscode.workspace.getConfiguration("Lua");
+	var value = baseLuaExtensionConfiguration.inspect("diagnostics.groupFileStatus");
+	if (value?.defaultValue === undefined
+		&& value?.globalValue === undefined
+		&& value?.workspaceValue === undefined
+		&& value?.workspaceFolderValue === undefined) {
```

Comment:
Not overwriting is good, but this is not set by default so you will overwrite this for any Lua project. This needs a check that this workspace is OpenRA or an SDK based mod. Check for existence of `mod.yaml` maybe?

Reply:
Good idea 👍